src/event_handler.rs

```diff
@@ -11,29 +11,37 @@ impl EventHandler {
 
     fn on_key_press(key: KeyEvent, app: &mut App) {
         match app.input_mode {
-            InputMode::Command => match key.code {
-                KeyCode::Esc | KeyCode::Char('q') | KeyCode::Char('Q') => app.is_running = false,
-                KeyCode::Char('d') => app.state.remove_selected_task(),
-                KeyCode::Char('x') => app.state.clear_all_tasks(), //Todo: add prompt for user ("Are you sure?...")
-                KeyCode::Up => app.state.tasks.previous(),
-                KeyCode::Down => app.state.tasks.next(),
-                KeyCode::Left => app.state.tasks.unselect(),
-                KeyCode::Char('a') | KeyCode::Char('A') => app.input_mode = InputMode::AddTask,
-                _ => (),
-            },
-            InputMode::AddTask => match key.code {
-                KeyCode::Enter => {
-                    app.state.add_task(&app.text_input);
-                    app.text_input = String::new();
-                    app.input_mode = InputMode::Command;
-                }
-                KeyCode::Backspace => {
-                    app.text_input.pop();
-                }
-                KeyCode::Esc => app.input_mode = InputMode::Command,
-                KeyCode::Char(c) => app.text_input.push(c),
-                _ => (),
-            },
+            InputMode::Command => Self::handle_command_input(key, app),
+            InputMode::AddTask => Self::handle_add_task_input(key, app),
+        };
+    }
+
+    fn handle_command_input(key: KeyEvent, app: &mut App) {
```

Comment:
I would be more specific with the name, maybe **handle_command_key_press**? i see potential conflict when we introduce for instance functions on_key_release, or on_mouse_click, and we also would like to separate them based on input mode